test/integration/model/sync.test.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+/* jshint -W030 */
+/* jshint -W110 */
+const chai = require('chai')
+  , Sequelize = require('../../../index')
+  , expect = chai.expect
+  , Support = require(__dirname + '/../support');
+
+describe(Support.getTestDialectTeaser('Model'), function() {
+  describe('sync', function() {
+    beforeEach(function() {
+      this.testSync = this.sequelize.define('testSync', {
+        dummy: Sequelize.STRING
+      });
+
+      return this.testSync.drop();
+    });
+
+    it('should remove a column if it exists in the databases schema but not the model', function() {
+      let User = this.sequelize.define('testSync', {
+        name: Sequelize.STRING,
+        age: Sequelize.INTEGER
+      });
+      return this.sequelize.sync()
+        .bind(this)
+        .then(() => {
+          this.sequelize.define('testSync', {
+            name: Sequelize.STRING
+          });
+        })
+        .then(() => this.sequelize.sync({alter: true}))
+        .then(() => {
+          return User.describe().then(data => {
+            expect(data).to.not.ownProperty('age');
+          });
+        });
+    });
+
+    it('should add a column if it exists in the model but not the database', function() {
+      let testSync = this.sequelize.define('testSync', {
+        name: Sequelize.STRING
+      });
+      return this.sequelize.sync()
+        .bind(this)
+        .then(() => {
+          this.sequelize.define('testSync', {
+            name: Sequelize.STRING,
+            age: Sequelize.INTEGER
+          });
+        })
+        .then(() => this.sequelize.sync({alter: true}))
+        .then(() => {
+          return testSync.describe().then(data => {
+            expect(data).to.ownProperty('age');
+          });
+        });
+    });
+
+    it('should change a column if it exists in the model but is different in the database', function() {
+      let testSync = this.sequelize.define('testSync', {
+        name: Sequelize.STRING,
+        age: Sequelize.INTEGER
+      });
+      return this.sequelize.sync()
+        .bind(this)
```

Comment:
removing this is causing the test to fail with 

`No description found for "testSyncs" table. Check the table name and schema; remember, they _are_ case sensitive.` and I don't really understand why.